include/up-cpp/communication/NotificationSource.h

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache License Version 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_CPP_COMMUNICATION_NOTIFICATIONSOURCE_H
+#define UP_CPP_COMMUNICATION_NOTIFICATIONSOURCE_H
+
+#include <up-cpp/datamodel/builder/Payload.h>
+#include <up-cpp/datamodel/builder/UMessage.h>
+#include <up-cpp/transport/UTransport.h>
+#include <uprotocol/v1/uattributes.pb.h>
+#include <uprotocol/v1/uri.pb.h>
+#include <uprotocol/v1/ustatus.pb.h>
+
+#include <chrono>
+#include <memory>
+#include <optional>
+#include <utility>
+
+namespace uprotocol::communication {
+/// @brief Interface for uEntities to receive notifications.
+///
+/// Like all L2 client APIs, the NotificationSource is a wrapper on top of the
+/// L1 UTransport API; in this instance, it provides for the notification
+/// sending half of the notification model.
+struct NotificationSource {
+	/// @brief Constructs a notification source connected to a given transport.
+	///
+	/// @post An internal UMessageBuilder will be configured based on the
+	///       provided attributes.
+	///
+	/// @param transport Transport to publish messages on.
+	/// @param sink URI of this uE. The authority and entity will be replaced
+	///             automatically with those found in the transport's default.
```

Comment:
By default, do you mean [this](https://github.com/eclipse-uprotocol/up-cpp/pull/105/files#diff-a36088a3d7aefa487d703f85584902256a8d27e3daec98f6cdaa6edced6d44a6R52)?

I'm unsure of a nicer way to tie these concepts together.

Reply:
Yes, but more specifically [getDefaultSource()](https://github.com/eclipse-uprotocol/up-cpp/pull/105/files#diff-a36088a3d7aefa487d703f85584902256a8d27e3daec98f6cdaa6edced6d44a6R119)